styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -120,14 +121,26 @@ public QueuedStateManager(
 
   @Override
   public CompletableFuture<Void> receive(Event event) throws IsClosedException {
+    // Read state counter at enqueueing time
```

Comment:
Should the default to _not_ check the counter at all and avoid additional Datastore lookups?

Reply:
Preliminary stress tests don't show issues with this additional lookup step in the StateManager